lib/datadog/appsec/component.rb

```diff
@@ -53,7 +53,7 @@ def reconfigure(ruleset:)
           if new && new.ready?
             old = @processor
             @processor = new
-            old.finalize
+            old.finalize if old
```

Comment:
Oh, right, `create_processor` can return `nil` (and also theoretically if `shutdown!` is called multiple times)

Reply:
While checking whether the processor being nil is properly checked ([which it is](https://github.com/DataDog/dd-trace-rb/blob/3c84e41991b5b176b685e0047d14a802164830b7/lib/datadog/appsec/contrib/rack/request_middleware.rb#L40)), I noticed we [create an immediately thrown away processor](https://github.com/DataDog/dd-trace-rb/blob/3c84e41991b5b176b685e0047d14a802164830b7/lib/datadog/appsec/contrib/rack/request_middleware.rb#L31-L33)...

This might have come from an automatic merge resolution, and line 31 should be removed.

@GustavoCaso can you fix that in this PR?